src/Client.js

```diff
@@ -876,11 +876,18 @@ class Client extends EventEmitter {
      * @returns {Promise<string>}
      */
     async getProfilePicUrl(contactId) {
-        const profilePic = await this.pupPage.evaluate((contactId) => {
+        const profilePic = await this.pupPage.evaluate(async contactId => {
             const chatWid = window.Store.WidFactory.createWid(contactId);
-            return window.Store.getProfilePicFull(chatWid);
+            let asyncPic = await window.Store.getProfilePicFull(chatWid).catch(() => {
+                return undefined;
+            });
+            if (!asyncPic) {
```

Comment:
I think we should be explicit and only call the Wap function if we're not on MD. 

So, this would look something like:

```
if(window.Store.Features.features.MD_BACKEND) {
// use getProfilePicFull
} else {
// use Wap.profilePicFind
}
```

Reply:
Sounds good! I'll apply those changes & test again